src/transformers/modeling_bart.py

```diff
@@ -939,8 +939,30 @@ def forward(
         return outputs
 
     @staticmethod
-    def prepare_inputs_for_generation(input_ids, past, **kwargs):
-        return {"input_ids": input_ids, "decoder_cached_states": past, "decoder_input_ids": input_ids}
+    def prepare_inputs_for_generation(decoder_input_ids, past, encoder_inputs, encoder_outputs, **wwargs):
```

Comment:
This is how I would implement the `prepare_inputs_for_generation` fn for Bart. What do you think? @sshleifer